cheroot/ssl/pyopenssl.py

```diff
@@ -150,14 +150,23 @@ def readline(self, size=-1):
             size,
         )
 
-    def sendall(self, *args, **kwargs):
-        """Send whole message to the socket."""
+    def read(self, *args, **kwargs):
+        """Read from the wrapped socket, with retry."""
         return self._safe_call(
-            False,
-            super(SSLFileobjectMixin, self).sendall,
+            True,
+            super(SSLFileobjectMixin, self).read,
             *args, **kwargs
         )
 
+    def sendall(self, *args, **kwargs):
+        """Send whole message to the socket.
+
+        Not supported due to https://github.com/pyca/pyopenssl/issues/176.
+        """
+        raise NotImplementedError(
+            'sendall() not supported on pyOpenSSL due to issue #176'
+        )
```

Comment:
Also, wouldn't it be cleaner if this was reimplemented using a series of `send()` calls in a loop?

Reply:
I suppose it would, but that would mean working around a third-party bug in a method that we don't actually use at all. I could also just remove the method, but I like that this was makes the non-support explicit.

Reply:
Looks like it's not really possible to do this now, after all.